pkg/operator/controllers/monitoring/monitoring_controller.go

```diff
@@ -54,52 +55,54 @@ type Config struct {
 	} `json:"alertmanagerMain,omitempty"`
 }
 
-type Reconciler struct {
-	log *logrus.Entry
-
-	client client.Client
+type MonitoringReconciler struct {
+	base.AROController
 
 	jsonHandle *codec.JsonHandle
 }
 
-func NewReconciler(log *logrus.Entry, client client.Client) *Reconciler {
-	return &Reconciler{
-		log:        log,
-		client:     client,
+func NewReconciler(log *logrus.Entry, client client.Client) *MonitoringReconciler {
+	return &MonitoringReconciler{
+		AROController: base.AROController{
+			Log:    log,
+			Client: client,
+			Name:   ControllerName,
+		},
 		jsonHandle: new(codec.JsonHandle),
 	}
 }
 
-func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
-	instance := &arov1alpha1.Cluster{}
-	err := r.client.Get(ctx, types.NamespacedName{Name: arov1alpha1.SingletonClusterName}, instance)
+func (r *MonitoringReconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	instance, err := r.GetCluster(ctx)
 	if err != nil {
 		return reconcile.Result{}, err
```

Comment:
Why not `SetDegraded` here? I noticed this is a convention in all of the controllers.

To me, it makes sense for the operator to report degraded as soon as it isn't able to Get the cluster CR for any reason. I'm curious why we decided to not do this. But if we decided to do it, it would probably be in a separate PR, so I'm good to approve this one.

Reply:
The simple reason is that these controller statuses are saved on the cluster CR itself, so if we're unable to retrieve it we'd be unable to set statuses anyways.